util/src/main/java/com/google/cloud/hadoop/util/ApiErrorExtractor.java

```diff
@@ -205,6 +207,18 @@ public boolean rateLimited(IOException e) {
     return false;
   }
 
+  /**
+   * Determines if a given Throwable is caused by Quota Exceeded. Recursively checks getCause() if
+   * outer exception isn't an instance of the correct class.
+   */
+  public boolean quotaExceeded(IOException e) {
```

Comment:
What status code GCS returns for requests with `quotaExceeded` reason? Is it `429`?

Reply:
I think the error code is 403, same as the rateLimitedExceeded

Reply:
`rateLimitedExceeded` has `429` code, I see that other methods in this class also check status code, so we may want to do the same for this new method too.

Reply:
ok, I will add a check for status code on this method

Reply:
I am not sure about whether the getReason is enough to determine the error type, since there is not a check for code for rateLimitedExceeded error.

Reply:
And also from [Google Cloud Error Messages](https://cloud.google.com/bigquery/docs/error-messages) the http code for both rateLimited and quotaExceeded error has the error code of 403.

Reply:
And from [Google Cloud Error Message](https://cloud.google.com/bigquery/docs/error-messages), both error have error code of 403.

Reply:
okay, let's leave it as is, and we will update all methods to check status code in the future.